### PR TITLE
ci: updates to run with node 22

### PR DESCRIPTION
**Pull request by big-kahuna-burger**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)
      
   `node-versions` can be passed as an argument as observed here:
   
   https://github.com/fastify/workflows/blob/main/.github/workflows/plugins-ci.yml#L34

### CODE REVIEW

**Comment by gurgunday** [comment]
Please see the _next_ branch

**Comment by big-kahuna-burger** [comment]
I see. So... change target to next and pin to 4.2.1?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ on:
 
 jobs:
   test:
-    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v3
+    uses: fastify/workflows/.github/workflows/plugins-ci.yml@main
```
Comment (mcollina) [inline .github/workflows/ci.yml:20]: please use the tagged version

**Review by mcollina — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**